### DISCUSSION TITLE
Prevent re-rendering of the component with stale data when refetching a query

### DISCUSSION THREAD

**Discussion by paulzakin** [TanStack/query#3927] [Q&A] [answered]
Let's say we have a (pseudo) component that looks like this. 

```tsx
// Query to fetch the data of shape { name: "foo" }  from the database
const { data } = useQuery(foo, async () => {...})
<input id="name" value={data.name} /> 
<Button onClick = () => {
// Save value of name in the database
db.save(name)
queryClient.refetchQueries("foo")
}>
Save
</Button>
```

The behaviour of the component is this:

-  We fetch the data on page load
-  The user changes the value of the input from "foo" to "bar".
-  The user clicks the Save button. 

As for as I understand, the following happens. 

- Render (First): The component renders once with the stale data. 
- This overwrites the value of "bar" in the state of the input with "foo".
- Render (Second): The query returns successfully and "bar" now obliterates "foo".

Is there any way to have the component only re-render with the refresh data? Basically, I want to skip Render (First)

**Comment by TkDodo** [comment, marked as answer]
if you're on v3, you can set `notifyOnChangeProps: 'tracked'` for your query so that it will only re-render if fields that you use / care about. Right now, it re-renders when anything changes, and `isFetching` is always changing.

in v4, this is the default behaviour.

see also: https://tkdodo.eu/blog/react-query-render-optimizations

**Reply by paulzakin** [reply]
Ok, great - thanks for the quick response!
